amount.go

```diff
@@ -360,7 +360,10 @@ func (a Amount) Value() (driver.Value, error) {
 // Allows scanning amounts from a PostgreSQL composite type.
 func (a *Amount) Scan(src interface{}) error {
 	// Wire format: "(9.99,USD)".
-	input := src.(string)
+	input, ok := src.(string)
+	if !ok {
+		return fmt.Errorf("invalid src: %v", src)
```

Comment:
This error does not tell us why the value is invalid.

What if we did this instead:
```
return fmt.Errorf("value is not a string: %v", src)
```